astropy/io/fits/hdu/hdulist.py

```diff
@@ -930,27 +936,30 @@ def info(self, output=None):
         else:
             name = self._file.name
 
-        results = ['Filename: {}'.format(name),
-                   'No.    Name      Ver    Type      Cards   Dimensions   Format']
-
-        format = '{:3d}  {:10}  {:3} {:11}  {:5d}   {}   {}   {}'
+        file_line = 'Filename: {}\n'.format(name)
+        results = []
         default = ('', '', '', 0, (), '', '')
         for idx, hdu in enumerate(self):
-            summary = hdu._summary()
-            if len(summary) < len(default):
-                summary += default[len(summary):]
-            summary = (idx,) + summary
+            hdu_summary = hdu._summary()
+            summary = [idx, *hdu_summary, *default[len(hdu_summary):]]
             if output:
-                results.append(format.format(*summary))
+                results.append(tuple(map(str, summary)))
             else:
-                results.append(summary)
+                results.append(tuple(summary))
 
         if output:
-            output.write('\n'.join(results))
+            t = Table(rows=results, names=COLUMN_NAMES)
```

Comment:
Maybe a little gain by giving columns directly (`Table` will do the `zip(*results)` but maybe with a few more - useless here - verifications).

Reply:
Might be worth a try 👍

Reply:
Not really faster (less than 1%), so I would prefer the easier to read option.